Sources/StreamChat/Controllers/MessageController/MessageController.swift

```diff
@@ -306,6 +328,68 @@ public extension ChatMessageController {
             self.callback { completion?(result.error) }
         }
     }
+
+    /// Loads the next page of reactions starting from the current fetched reactions.
+    ///
+    /// - Parameters:
+    ///   - limit: The reactions page size.
+    ///   - completion: The completion is called when the network request is finished.
+    ///   If the request fails, the completion will be called with an error, if it succeeds it is
+    ///   called without an error and the delegate is notified of reactions changes.
+    func loadNextReactions(
+        limit: Int = 25,
+        completion: ((Error?) -> Void)? = nil
+    ) {
+        if hasLoadedAllReactions {
+            callback { completion?(nil) }
+            return
+        }
+
+        loadReactions(limit: limit, offset: reactions.count) { result in
+            switch result {
+            case let .success(reactions):
+                let currentReactions = Set(self.reactions)
+                for reaction in reactions where !currentReactions.contains(reaction) {
+                    self.reactions.append(reaction)
+                }
+
+                if reactions.count < limit {
+                    self.hasLoadedAllReactions = true
+                }
+
+                completion?(nil)
+
+            case let .failure(error):
+                completion?(error)
+            }
+        }
+    }
+
+    /// Loads reactions from the backend given an offset and a limit.
+    ///
+    /// - Parameters:
+    ///   - limit: The reactions page size.
+    ///   - offset: The starting position from the desired range to be fetched.
+    ///   - completion: The completion is called when the network request is finished.
+    ///   It is called with the reactions if the request succeeds or error if the request fails.
+    func loadReactions(
+        limit: Int,
+        offset: Int = 0,
+        completion: @escaping (Result<[ChatMessageReaction], Error>) -> Void
+    ) {
+        messageUpdater.loadReactions(
+            cid: cid,
+            messageId: messageId,
+            pagination: Pagination(pageSize: limit, offset: offset)
+        ) { result in
+            switch result {
+            case let .success(reactions):
+                self.callback { completion(.success(reactions)) }
+            case let .failure(error):
+                self.callback { completion(.failure(error)) }
```

Comment:
I wouldn't hide switching to the callback queue in this method but rather do it explicitly where needed. 

The thing is we don't have a reliable way to check on which queue we currently are we have the issue with nested `self.callback { ... }` statements: if the callback queue is `background` and `serial` we get a deadlock:
```swift
extension Controller {
    /// A helper function to ensure the callback is performed on the callback queue.
    func callback(_ action: @escaping () -> Void) {
        if Thread.current.isMainThread {
            if callbackQueue == .main {
                // We perform the callback synchronously
                action()
            } else {
                // We dispatch from the main queue, we must perform
                // the callback must be performed asynchronously
                callbackQueue.async {
                    action()
                }
            }
        } else {
            // Dispatching from a background queue, the callback can be performed synchronously
            callbackQueue.sync { <- here we get a deadlock
                action()
            }
        }
    }
}
```

Getting back to the code, we switch to callback queue in `loadReactions` and then again in `reactions.didSet` when accessing the delegate so for with serial bg callback queue it will 💥